### PR TITLE
test: added tests for dropEmptyAssays()

### PR DESCRIPTION
**Pull request by cvanderaa**
Note that I added one minor line to the function to fulfill an edge case.

### CODE REVIEW

File: R/QFeatures-class.R [hunk 1]
```diff
@@ -1157,5 +1157,6 @@ dropEmptyAssays <- function(object, dims = 1:2) {
         object <- object[, , nrows(object) > 0]
     if (2 %in% dims)
         object <- object[, , ncols(object) > 0]
+    if (!length(object)) return(QFeatures())
```
Comment (lgatto) [inline R/QFeatures-class.R:1160]: Should you not start by this case?

Comment (cvanderaa) [reply R/QFeatures-class.R:1160]: No, the edge case I want to solve happens after the subsetting. For example, suppose all assays have no features. `dropEmptyAssays()` will remove all assays and return an empty `Qfeatures` object. Without this line, the object is empty, but there is some leftover information, like metadata or colnames in the empty colData.

As a note, the function was working perfectly for `dropEmptyAssays(QFeatures())`.

Comment (cvanderaa) [reply R/QFeatures-class.R:1160]: Btw, maybe this case could deserve a warning?

Comment (lgatto) [reply R/QFeatures-class.R:1160]: Thanks. 

No, I wouldn't throw a warning, because it would be the normal/expected behaviour. A message at best, but I'm not even convinced that would be needed.
